dev/build-example-docs.py

```diff
@@ -0,0 +1,236 @@
+    readme_file = os.path.join(ROOT, "examples", "doc", "source", example + ".md")
+    with open(readme_file, "r+") as f:
+        content = f.read()
+        if gh_text not in content:
+            content = re.sub(
+                r"(^# .+$)", rf"\1\n\n{gh_text}", content, count=1, flags=re.MULTILINE
+            )
+            f.seek(0)
+            f.write(content)
+            f.truncate()
+
+
+def _copy_images(example):
+    static_dir = os.path.join(example, "_static")
+    dest_dir = os.path.join(ROOT, "examples", "doc", "source", "_static")
+    if os.path.isdir(static_dir):
+        for file in os.listdir(static_dir):
+            if file.endswith((".jpg", ".png", ".jpeg")):
+                shutil.copyfile(
+                    os.path.join(static_dir, file), os.path.join(dest_dir, file)
+                )
+
+
+def _add_all_entries(index_file):
+    examples_dir = os.path.join(ROOT, "examples")
+    for example in sorted(os.listdir(examples_dir)):
+        example_path = os.path.join(examples_dir, example)
+        if os.path.isdir(example_path) and example != "doc":
+            _copy_markdown_files(example_path)
+            _add_gh_button(example)
+            _copy_images(example)
+
+
+def _main():
+    if os.path.exists(INDEX):
+        os.remove(INDEX)
+
+    with open(INDEX, "w") as index_file:
+        index_file.write(initial_text)
+
+    examples_dir = os.path.join(ROOT, "examples")
+    for example in sorted(os.listdir(examples_dir)):
+        example_path = os.path.join(examples_dir, example)
+        if os.path.isdir(example_path) and example != "doc":
+            _copy_markdown_files(example_path)
+            _add_gh_button(example)
+            _copy_images(example_path)
+            if not _add_table_entry(example_path, "quickstart", "quickstart"):
+                if not _add_table_entry(example_path, "comprehensive", "comprehensive"):
+                    if not _add_table_entry(example_path, "advanced", "advanced"):
+                        _add_table_entry(example_path, "", "other")
+
+    with open(INDEX, "a") as index_file:
+        index_file.write(categories["quickstart"]["table"])
+
+        index_file.write("\nAdvanced Examples\n-----------------\n")
+        index_file.write(
+            "Advanced Examples are mostly for users that are both familiar with "
+            "Federated Learning but also somewhat familiar with Flower's main "
+            "features.\n"
+        )
+        index_file.write(categories["advanced"]["table"])
+
+        index_file.write("\nOther Examples\n--------------\n")
+        index_file.write(
+            "Flower Examples are a collection of example projects written with "
+            "Flower that explore different domains and features. You can check "
+            "which examples already exist and/or contribute your own example.\n"
```

Comment:
should we add a link to the "contribute your own example" text pointing to our contributors guide?